src/mscorlib/src/System/IO/TextWriter.cs

```diff
@@ -606,30 +606,69 @@ public virtual Task FlushAsync()
         [Serializable]
         private sealed class NullTextWriter : TextWriter
         {
-            internal NullTextWriter(): base(CultureInfo.InvariantCulture) {
-            }
-        
-            public override Encoding Encoding {
-                get { return Encoding.Default;  }
+            internal NullTextWriter() : base(CultureInfo.InvariantCulture)
+            {
             }
 
-            [SuppressMessage("Microsoft.Contracts", "CC1055")]  // Skip extra error checking to avoid *potential* AppCompat problems.
-            public override void Write(char[] buffer, int index, int count) {
-            }
-            
-            public override void Write(String value) {
-            }
+            public override Encoding Encoding => Encoding.Unicode;
 
-            // Not strictly necessary, but for perf reasons
-            public override void WriteLine() {
-            }
+            public override Task FlushAsync() => Task.CompletedTask;
 
-            // Not strictly necessary, but for perf reasons
-            public override void WriteLine(String value) {
-            }
-            
-            public override void WriteLine(Object value) {
+            [SuppressMessage("Microsoft.Contracts", "CC1055")]  // Skip extra error checking to avoid *potential* AppCompat problems.
+            public override void Write(char[] buffer, int index, int count)
+            {
             }
+
+            // These overrides for Write and WriteLine
+            // aren't strictly necessary, but we do so
+            // for perf reasons (nooping instead of e.g.
+            // allocating a new string in Write(int),
+            // which by default calls int.ToString).
+
+            public override void Write(bool value) { }
+            public override void Write(char value) { }
+            public override void Write(char[] buffer) { }
+            public override void Write(decimal value) { }
+            public override void Write(double value) { }
+            public override void Write(float value) { }
+            public override void Write(int value) { }
+            public override void Write(long value) { }
+            public override void Write(object value) { }
+            public override void Write(string format, object arg0) { }
+            public override void Write(string format, object arg0, object arg1) { }
+            public override void Write(string format, object arg0, object arg1, object arg2) { }
+            public override void Write(string format, params object[] arg) { }
```

Comment:
@jkotas OK, will revert the changes in this file then (as they take up a lot of code as well).